### PR TITLE
Add REORG operation details in commit history

### PR DESCRIPTION
**Pull request by vkorukanti**
## Description
`REORG` command was added as part of the delta-io/delta#1732. The operation is recorded as `OPTIMIZE`. This PR changes it to `REORG`

## How was this patch tested?
Updated existing UTs to verify the Delta table history

### CODE REVIEW

**Comment by vkorukanti** [comment]
@xupefei Please review.

File: core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala [hunk 1]
```diff
@@ -108,6 +123,31 @@ class DeltaReorgSuite extends QueryTest
...
+      tablePath: String,
+      expOpParams: Map[String, String],
+      numFilesRemoved: Long,
+      numFilesAdded: Long): Unit = {
+    val (opName, opParams, opMetrics) = DeltaTable.forPath(tablePath)
+      .history(1)
+      .select("operation", "operationParameters", "operationMetrics")
+      .as[(String, Map[String, String], Map[String, String])]
+      .head()
+    assert(opName === "REORG")
```
Comment (allisonport-db) [inline core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala:146]: Do we check for opName in the optimize suites?

Comment (vkorukanti) [reply core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala:146]: We are [not](https://github.com/delta-io/delta/blob/master/core/src/test/scala/org/apache/spark/sql/delta/optimize/OptimizeMetricsSuite.scala). Will followup with a separate PR.

**Comment by xupefei** [comment]
LGTM, thanks for the work!

File: core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala (hunk 1 above)
Comment (vkorukanti) [reply core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala:146]: https://github.com/delta-io/delta/issues/1756

**Review by allisonport-db — APPROVED** [review]
LGTM

File: core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala (hunk 1 above)
Comment (felipepessoto) [reply core/src/test/scala/org/apache/spark/sql/delta/optimize/DeltaReorgSuite.scala:146]: https://github.com/delta-io/delta/pull/1758

**Pull request closed without merging**